src/bespokelabs/curator/request_processor/online_request_processor.py

```diff
@@ -0,0 +1,312 @@
+        status_tracker: StatusTracker,
+    ) -> None:
+        """Process a single API request"""
+        pass
+
+    def check_structured_output_support(self, prompt_formatter: PromptFormatter) -> bool:
+        """Check if the model supports structured output"""
+        return True
+
+    def run(
+        self,
+        dataset: Optional[Dataset],
+        working_dir: str,
+        parse_func_hash: str,
+        prompt_formatter: PromptFormatter,
+    ) -> Dataset:
+        """Run completions using the online API with async processing."""
+        logger.info(f"Running {self.__class__.__name__} completions with model: {self.model}")
+
+        self.prompt_formatter = prompt_formatter
+        if self.prompt_formatter.response_format:
+            assert self.check_structured_output_support(
+                prompt_formatter
+            ), f"Model {self.model} does not support structured output, response_format: {self.prompt_formatter.response_format}"
+        generic_requests_files = self.create_request_files(dataset, working_dir, prompt_formatter)
+        generic_responses_files = [
+            f"{working_dir}/responses_{i}.jsonl" for i in range(len(generic_requests_files))
+        ]
+
+        for request_file, response_file in zip(generic_requests_files, generic_responses_files):
+            run_in_event_loop(
+                self.process_requests_from_file(
+                    generic_requests_filepath=request_file,
+                    save_filepath=response_file,
+                    max_attempts=5,
+                    resume=True,
+                )
+            )
+
+        return self.create_dataset_files(working_dir, parse_func_hash, prompt_formatter)
+
+    async def process_requests_from_file(
+        self,
+        generic_requests_filepath: str,
+        save_filepath: str,
+        max_attempts: int,
+        resume: bool,
+    ) -> None:
+        """Processes API requests in parallel, throttling to stay under rate limits."""
+
+        # Initialize trackers
+        queue_of_requests_to_retry: asyncio.Queue[APIRequest] = asyncio.Queue()
+        status_tracker = StatusTracker()
+
+        # Get rate limits
+        rate_limits = self.get_rate_limits()
+        status_tracker.max_requests_per_minute = rate_limits["max_requests_per_minute"]
+        status_tracker.max_tokens_per_minute = rate_limits["max_tokens_per_minute"]
+        rpm = rate_limits["max_requests_per_minute"]
+
+        soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+        resource.setrlimit(
+            resource.RLIMIT_NOFILE,
+            (min(hard, int(10 * status_tracker.max_requests_per_minute)), hard),
+        )
+
+        # Track completed requests for resume functionality
+        completed_request_ids = set()
+        if os.path.exists(save_filepath) and resume:
+            with open(save_filepath, "r") as f:
+                for line in f:
+                    response = GenericResponse.model_validate_json(line)
```

Comment:
add logging back in to say we are resuming from the saved file and how many are completed

Reply:
Also the logic to write to a temporary file that filters out the error'd requests for retrial and then swap the temporary file back in https://github.com/bespokelabsai/curator/blob/main/src/bespokelabs/curator/request_processor/openai_online_request_processor.py#L266-L268

Reply:
brought back